nil/services/txnpool/txnpool.go

```diff
@@ -110,10 +113,16 @@ func (p *TxnPool) listen(ctx context.Context, sub *network.Subscription) {
 }
 
 func (p *TxnPool) Add(ctx context.Context, txns ...*types.Transaction) ([]DiscardReason, error) {
+	if len(txns) == 0 {
+		return nil, nil
+	}
 	mms := make([]*metaTxn, len(txns))
+
+	p.lock.Lock()
 	for i, txn := range txns {
 		mms[i] = newMetaTxn(txn, p.baseFee)
 	}
+	p.lock.Unlock()
```

Comment:
Is it possible to add protected getter for this field?

```suggestion

func (p *TxnPool) GetBaseFee() (baseFee types.Value) {
    return p.baseFee
}

func (p *TxnPool) Add(ctx context.Context, txns ...*types.Transaction) ([]DiscardReason, error) {
	if len(txns) == 0 {
		return nil, nil
	}
	mms := make([]*metaTxn, len(txns))

	baseFee := p.GetBaseFee()
	for i, txn := range txns {
		mms[i] = newMetaTxn(txn, baseFee)
	}
```